cmake/toolchain/zephyr/host-tools.cmake

```diff
@@ -7,51 +7,77 @@
 # and currently provided by the Zephyr SDK. Those tools will need to be
 # provided for different OSes and sepearately from the toolchain.
 
-set_ifndef(ZEPHYR_SDK_INSTALL_DIR "$ENV{ZEPHYR_SDK_INSTALL_DIR}")
-set(ZEPHYR_SDK_INSTALL_DIR ${ZEPHYR_SDK_INSTALL_DIR} CACHE PATH "Zephyr SDK install directory")
+set(REQUIRED_SDK_VER 0.11.1)
+
+set_ifndef(ZEPHYR_TOOLCHAIN_VARIANT $ENV{ZEPHYR_TOOLCHAIN_VARIANT} "")
+set_ifndef(ZEPHYR_SDK_INSTALL_DIR   $ENV{ZEPHYR_SDK_INSTALL_DIR} "")
+
+# There are three scenarios where Zephyr SDK should be looked up:
+# 1) Zephyr specified as toolchain (ZEPHYR_SDK_INSTALL_DIR still used if defined)
+# 2) No toolchain specified == Default to Zephyr toolchain (Linux only)
+# 3) Other toolchain specified, but ZEPHYR_SDK_INSTALL_DIR also given.
+#    This means Zephyr SDK toolchain will not be used for compilation,
+#    but other supplementary host tools will be used.
+if(("zephyr" STREQUAL ${ZEPHYR_TOOLCHAIN_VARIANT}) OR
+   ((NOT DEFINED ZEPHYR_TOOLCHAIN_VARIANT) AND (${CMAKE_HOST_SYSTEM_NAME} STREQUAL Linux)) OR
+   (DEFINED ZEPHYR_SDK_INSTALL_DIR))
+
+  # No toolchain was specified, so inform user that we will be searching.
+  if (NOT DEFINED ZEPHYR_SDK_INSTALL_DIR AND NOT DEFINED ZEPHYR_TOOLCHAIN_VARIANT)
+    message("ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK")
+  endif()
+
+  # This ensure packages are sorted in ascending order.
+  # CMake will select the first matching package, and thus the version closest
+  # to the required, as example, requesting 0.11.1:
+  # Zephyr-sdk: 0.10.0  (Not matching)
+  # Zephyr-sdk: 0.11.1  (First matching - will be used)
+  # Zephyr-sdk: 0.12.0  (Matching)
```

Comment:
@galak and @nashif if the required version is 0.11.1, and the user has the SDK versions installed above an has not set `ZEPHYR_SDK_INSTALL_DIR`, should we use 0.11.1 or 0.12.0?

Reply:
I have no strong preference, but the reason I chose this order is that whenever we release Zephyr, then we are testing with the minimum required version, as that will usually be the only working SDK available at the time of release.
This again means this particular SDK version is assumed to be fully working with this Zephyr release.

When we release a new SDK (for example 0.11 -> 0.12), then that version is of course expected to be backward compatible, but as we have not fully tested all corners of a new SDK with the old release, then I decided the safest bet was to select the closest version when multiple versions are available.

But choosing newest compatible version is also good choice.
I just had to present one way or the other in this PR, and the reason above states why this is the initial proposal.

Reply:
Dev-review: Use the latest version instead of the required one.

Reply:
fixed